### ISSUE TITLE
Crash updating UI on background thread

### ISSUE THREAD

**Issue by pjrobertson** [quicksilver/Quicksilver#2668]
**Describe the bug**
Xcode gives a `Unexpected outstanding background CATransaction` message and a crash when opening the UI while also checking for updates.

**To Reproduce**
1. Turn on 'Check for Updates' → 'On Launch'
2. Restart Quicksilver
3. Immediately open up the preferences, and then scroll on any scroll view
4. See Quicksilver Crash

**Screenshots**

https://user-images.githubusercontent.com/150431/157686741-264db3cc-ff5c-4331-b4b8-37d852ba9b54.mov

**What version of macOS?**
10.14.6

**What version of Quicksilver?**
Master + 1.6.1 
I thought it was because of my recent changes to add a spinning wheel in the UI to show when updating, but I went back to the current release and I still got the same problem.

**Comment by skurfer** [comment]
I remember consistently seeing a crash when checking for updates because it was trying to show the alert window on a background thread. I thought we fixed it, but can’t find the PR. Maybe it was fixed in an OS update?

In any case, I can’t reproduce the crash using either of the methods described here.

**Comment by pjrobertson** [comment]
Reopening as I found out the reason (plugin updates were running on a background thread and then causing issues when updating the interface).

Fixed by #2671

**Comment by pjrobertson** [comment]
Fixed by #2671

**Comment by pjrobertson** [comment]
Thanks for the debugging.

OK, so I think this is a macOS bug then. I debugged, and the NSAlert *is* getting called on the main thread, but what’s happening is this:

1. NSAlert gets called on main thread, blocking it
2. The OS decides to use NSConcurrentScreenUpdate to update the UI (I can’t find a way not to do this)
3. The OS tried to do an update on the background thread
4. Boom

I’ve only noticed this bug now because I turned on ‘Check for updates on launch’ to debug some other stuff. I think this is a pretty rare edge case, so I won’t spend any more time on it.
